shell/platform/common/cpp/text_input_model.cc

```diff
@@ -91,9 +93,10 @@ bool TextInputModel::Backspace() {
     DeleteSelected();
     return true;
   }
-  if (selection_base_ != text_.begin()) {
-    int count = IsTrailingSurrogate(*(selection_base_ - 1)) ? 2 : 1;
-    selection_base_ = text_.erase(selection_base_ - count, selection_base_);
+  if (selection_base_ != 0) {
+    int count = IsTrailingSurrogate(text_.at(selection_base_ - 1)) ? 2 : 1;
+    text_.erase(selection_base_ - count, count);
+    selection_base_ -= count;
```

Comment:
Are backspace and delete handling reversed selections correctly? This looks suspicious (not just the updated versions, but in general).

Reply:
If selection base and extent differ we bail out above, so base==extent here. I ended up adding comments to this effect in an upcoming PR since every time I look at this I end up squinting at it until I spot that. I've updated this PR to include those comments.